lib/frontend/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+set(LLVM_LINK_COMPONENTS
+  Option
+  Support
+  )
+
+
+llvm_add_library(flangFrontend
+  TextDiagnosticPrinter.cpp
+
+  LINK_LIBS
+  clangBasic
+  clangDriver
+  )
+
+include_directories(${LLVM_DIR} ${LLVM_DIR}/../..)
+
+target_include_directories(flangFrontend
+  PUBLIC
+  ${LLVM_DIR}/../..)
+
+set_target_properties(flangFrontend
+  PROPERTIES
+  COMPILE_FLAGS -Wno-unused-parameter)
```

Comment:
If I recall correctly, because flang's default -Werror/-Wall flags turn it on, and the code as taken from flang was firing the warning (and therefore an error). I agree it should be removed and the related warnings fixed as necessary.

Reply:
I think there is also code in some llvm headers that trigger this warning, so we will probably have to turn it off eventually. I don't think it's very useful and we should turn it off for all of f18.

Reply:
I have found the option to be useful and would like to retain it if possible.